### ISSUE TITLE
Specifying bio_sample_id in SearchReadGroupSets requests is awkward and inconsistent

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/schemas#650]
The current definitions of SearchReadGroupSetsRequest has the following:
```javascript
  //  Specifying the id of a BioSample record will return only readgroups
  //  with the given bioSampleId.
  string bio_sample_id = 3;
```

This is currently interpreted in the reference server to mean that (a) we should only return read group sets that contain read groups that refer to this ``bio_sample_id`` and (b) for these read groups sets, we filter out all read groups that do not refer to this ``bio_sample_id``. This is semantically very awkward, and inconsistent with other parts of the API. 

It would be far simpler if we had a ``SearchReadGroupsRequest`` which provided a set of filters on the ReadGroups (including bio_sample_id) and returned a list of ReadGroup objects.

**Comment by david4096** [comment]
Agreed! Having a read groups endpoint helps to make the API consistent. Let me know what you think of #652.

**Comment by jeromekelleher** [comment]
Looks good to me @david4096, a rethink of the reads API is long overdue IMO.
